persistence/src/test/java/cz/muni/fi/pa165/HouseDaoTest.java

```diff
@@ -31,68 +39,60 @@
     public EntityManager em;
 
     @Autowired
-    public HouseDao houseDao;
+    private HouseDao houseDao;
 
-    private House h1;
-    private House h2;
+    private House house;
 
 
     @BeforeMethod
     public void createHouses() {
-
-        h1 = new House();
-        h2 = new House();
-        h1.setName("SCALA");
-        h2.setName("MORGAL");
-
+        house = new House()
+                .setName("SCALA")
+                .setAddress("MORAVAK")
+                .setHistory("Brief history of time")
+                .setHauntingFrom(Calendar.getInstance().getTime());
 
     }
+
     @Test
     public void createHouseTest() {
-        assertThat(h1.getId()).isNull();
-        houseDao.create(h1);
-        assertThat(h1.getId()).isNotNull();
+        assertThat(house.getId()).isNull();
+        houseDao.create(house);
+        assertThat(house.getId()).isNotNull();
+
     }
 
     @Test
-    public void deleteHouseTest() {
-        houseDao.create(h1);
-        houseDao.delete(h1);
-
-        assertThat(houseDao.getByName("SCALA")).isNull();
+    public void updateTest(){
+        houseDao.create(house);
+        house.setName("MORGAL");
+        House updated = houseDao.update(house);
+        assertThat(houseDao.getByName("MORGAL")).isEqualTo(updated);
     }
 
     @Test
-    public void getByNameTest() {
-        houseDao.create(h1);
-        assertThat(houseDao.getByName("SCALA")).isEqualTo(h1);
+    public void deleteTest(){
+        houseDao.create(house);
+        houseDao.delete(house);
+        assertThat(houseDao.getAll()).hasSize(0);
     }
 
     @Test
-    public void updateHouseTest() {
-        houseDao.create(h1);
-        House h = houseDao.update(h1.setAddress("MORAVAK"));
-        assertThat(houseDao.getByAddress("MORAVAK")).isEqualTo(h);
+    public void getAllTest(){
+        houseDao.create(house);
+        assertThat(houseDao.getAll()).hasSize(1);
     }
 
     @Test
-    public void getAllTest() {
-        houseDao.create(h1);
-        houseDao.create(h2);
-
-        List<House> all = houseDao.getAll();
-
-        assertThat(all.size()).isEqualTo(2);
-        assertThat(all.contains(h1));
-        assertThat(all.contains(h2));
-
+    public void getByNameTest(){
+        houseDao.create(house);
+        assertThat(houseDao.getByName("SCALA")).isEqualTo(house);
```

Comment:
to to ak si das do Before metody tak to potom si zavolas house1.getName(), nepouzivaj hardcoded string